certificate-service/pdfbox-generator/src/main/java/se/inera/intyg/certificateservice/pdfboxgenerator/pdf/value/PdfElementValueGenerator.java

```diff
@@ -1,14 +1,40 @@
 package se.inera.intyg.certificateservice.pdfboxgenerator.pdf.value;
 
-import java.io.IOException;
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 import se.inera.intyg.certificateservice.domain.certificate.model.Certificate;
 import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.PdfValueType;
 import se.inera.intyg.certificateservice.pdfboxgenerator.pdf.PdfField;
 
-public interface PdfElementValueGenerator {
+@Service
+@RequiredArgsConstructor
+public class PdfElementValueGenerator {
 
-  List<PdfField> generate(Certificate certificate, ElementId id, String fieldId)
-      throws IOException;
+  private final List<PdfElementValue> pdfElementValues;
 
+  public List<PdfField> getFields(Certificate certificate) {
```

Comment:
Should this method be called generate, and the already defined generate method maybe should be called getFields?

I think it better describes both methods?